assignments/array-methods.js

```diff
@@ -56,21 +56,48 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
-console.log(fullName);
+
+runners.forEach(function(element){
+    fullName.push(element.first_name);
+    fullName.push(element.last_name);
+
+});
+
+console.log(fullName);  
+
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+
+runners.map(function(currentValue){
+    return allCaps.push(currentValue.first_name.toUpperCase());
+
+});
+
+
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+runners.filter((currentValue) => {
+    return largeShirts.push(currentValue.shirt_size === "L");
+});
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+
+runners.reduce(function(accumilator, currentValue){
+
+    return ticketPriceTotal.push(accumilator + currentValue.donation);
+}, 0); // starting point - starts from 0
+
+
```

Comment:
Good job. FYI the accumulator is a 'foo bar' thing (i.e. you can name it whatever you wish).